staticaddr/withdraw/manager.go

```diff
@@ -325,20 +340,32 @@ func (m *Manager) WithdrawDeposits(ctx context.Context,
 		deposits, allWithdrawing = m.cfg.DepositManager.AllOutpointsActiveDeposits(
 			outpoints, deposit.Withdrawing,
 		)
-
 		if !allWithdrawing {
 			return "", "", ErrWithdrawingMixedDeposits
 		}
+	}
+
+	// Lock the deposits to prevent concurrent deposit access.
+	lockDeposits(deposits)
+	locked := true
+	defer func() {
+		if locked {
```

Comment:
nit: we could set `locked = true` in `lockDeposits` and then check `if locked` within `unlockDeposits` for a slight readability gain.